### PR TITLE
Auto save

### PR DESCRIPTION
**Pull request by Roeterdink**
Add optional auto save function, including menu setting for selection of save interval time, see thread :
https://www.elvastower.com/forums/index.php?/topic/35870-auto-save/page__view__findpost__p__280206

### CODE REVIEW

**Comment by Looky1173** [comment]
@twpol For some reason, this PR by Rob doesn't get picked up by the OR bot despite the `for-unstable` label.

**Comment by SteelFill** [comment]
Looks like this is conflicting with something in Viewer.cs (go figure, a quarter of PRs make changes in there) but I don't want to see it sit around forever because this is a good feature to have. I'll build this locally and try it out, and maybe figure out what other PR is causing the conflict.

**Comment by twpol** [comment]
I have removed #886 from Unstable as I think it is the one blocking this PR being included.

**Review by SteelFill — DISMISSED** [review]
Great feature to have, the implementation makes sense and seems to work as intended in my testing over the last couple days. Autosaves correctly happen at the expected intervals in real time, pausing the game suppresses autosave until unpaused, manual saving resets the autosave timer, and the buttons in the options menu do what they should.

However, for consistency this feature needs the "i" icon next to the Auto Save checkbox in the options menu, with a link to a feature description in the manual (also missing).


Also, I don't consider it necessary, but if it's easy to implement I'd appreciate an option to set the maximum number of autosaves allowed, automatically deleting the oldest autosave as a new one is made to remain within the limit. A lot of games do that for good reason, it would help save some space and reduce babysitting of the savegame list.

**Comment by Roeterdink** [comment]
As my last large update to the timetable code could not be merged with the unstable version due to an excessive amount of merge conflicts, I was forced to create my own version of the code. This version is now no longer in any way compatible with any of the 'official' versions. As a result, I no longer have a code base from which I can make valid commits of any changes. I do not intend to, again, spend a lot of time trying to sort out this problem. So I can also not make the required changes to the code for this PR. In fact, in all I do no longer see myself as an active OR code developer.
I can leave this PR for now to see if anyone else is willing to pick this up. But if that does not happen, I think it is best if this PR is removed. A PR which requires changes which nobody is willing to make, and which therefor can never be approved, is of no use to anyone. I will leave this for another few weeks, but if noone shows any interest by that time I will remove it.

Regards,
Rob Roeterdink

**Comment by SteelFill** [comment]
I see, that stinks. I'm personally interested in this feature so I'm willing to recreate it in my repo and start off from there (which would require closing this PR and opening a new one), but I can't help ya on timetable code.

**Comment by cjakeman** [comment]
Label changed to not-for-unstable temporarily to hide changes to UserSettings.cs to let other PRs merge into Unstable

**Comment by cjakeman** [comment]
And restored again

**Comment by cjakeman** [comment]
Marking this as not-for-unstable so that it no longer conflicts with PR980.
This is temporary - just for this weekend.

File: Source/RunActivity/Viewer3D/Viewer.cs [hunk 1]
```diff
@@ -76,6 +76,7 @@ public class Viewer
         /// Monotonically increasing time value (in seconds) for the game/viewer. Starts at 0 and only ever increases, at real-time.
         /// </summary>
         public double RealTime { get; private set; }
+        public double LastSave = -1;
```
[the remark this answers is not in the document]
Comment (cjakeman) [reply Source/RunActivity/Viewer3D/Viewer.cs]: I've changed it to AutoSaveDueAt

**Review by cjakeman — APPROVED** [review]
Hi Rob,

Thanks for this. I've added the missing section for the Manual.

Approved and merged.

Chris

**Pull request merged**